jib-core/src/main/java/com/google/cloud/tools/jib/registry/credentials/json/DockerConfigTemplate.java

```diff
@@ -118,37 +137,19 @@ public String getAuthFor(String registry) {
    */
   @Nullable
   public String getCredentialHelperFor(String registry) {
-    if (credsStore != null) {
-      // The registry could be prefixed with the HTTPS protocol and/or have suffixes.
-      Predicate<String> registryMatch =
-          key -> key.startsWith(registry) || key.startsWith("https://" + registry);
-      if (auths.keySet().stream().anyMatch(registryMatch)) {
-        return credsStore;
-      }
+    List<Predicate<String>> registryMatchers = getRegistryMatchersFor(registry);
+    if (credsStore != null && findFirstInMapByKey(auths, registryMatchers) != null) {
+      return credsStore;
     }
-    return findFirstInMapByKey(credHelpers, getRegistryMatchersFor(registry));
+    return findFirstInMapByKey(credHelpers, registryMatchers);
   }
 
-  private Stream<Predicate<String>> getRegistryMatchersFor(String registry) {
+  private List<Predicate<String>> getRegistryMatchersFor(String registry) {
```

Comment:
Hmm what was the reason for changing this to a List?

Reply:
Can't reuse the stream once it is closed with `Steam.firstFirst()`. That is,

```java
Stream<...> matchers = getRegistryMatchersFor(...);
findFirstInMapByKey(..., matchers);  // first call OK
findFirstInMapByKey(..., matchers);  // <-- throws an exception
```

Reply:
Ah right